File: app/src/main/java/com/example/seprojectsemester5/SelectRole.java
package com.example.seprojectsemester5;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class SelectRole extends AppCompatActivity {
    CardView collectorCard,analystCard;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_select_role);
        collectorCard = findViewById(R.id.collectorCard);
        analystCard = findViewById(R.id.analystCard);
        collectorCard.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(SelectRole.this,Login.class));
                finish();
            }
        });
        analystCard.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(SelectRole.this,AnalystLogin.class));
                finish();
            }
        });
    }
}

File: app/src/main/java/com/example/seprojectsemester5/SelectRole.kt
package com.example.seprojectsemester5

import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import android.os.Bundle
import android.content.Intent
import com.example.seprojectsemester5.dataanalyst.DataAnalystMainActivity

class SelectRole : AppCompatActivity() {
    private lateinit var collectorCard: CardView
    private lateinit var analystCard: CardView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_select_role)
        collectorCard = findViewById(R.id.collectorCard)
        analystCard = findViewById(R.id.analystCard)

        collectorCard.setOnClickListener {
            startActivity(Intent(this@SelectRole, Login::class.java))
            finish()
        }
        analystCard.setOnClickListener { // bypassing the authenticating right now
            startActivity(Intent(this@SelectRole, DataAnalystMainActivity::class.java))
            // startActivity(new Intent(SelectRole.this, AnalystLogin.class));
            finish()
        }
    }
}